gc/src/trace.rs

```diff
@@ -135,3 +203,41 @@ unsafe impl<T: Trace> Trace for Option<T> {
         }
     });
 }
+
+impl<K: Trace, V: Trace> Finalize for BTreeMap<K, V> {}
+unsafe impl<K: Trace, V: Trace> Trace for BTreeMap<K, V> {
+    custom_trace!(this, {
+        for (k, v) in this {
```

Comment:
Why not just `this.iter().map(mark)`? Shouldn't that work now due to the tuple implementations?

Reply:
Iterators are lazy, so I'd need to demand the value at some point; I felt a for-loop would be clearer.
Now that I do have the tuple impls, though, I'll change it to

```rust
for pair in this {
    mark(pair);
}
```

Reply:
Ach, cancel that, there's lifetime issues (`this` is borrowed, not owned?).

Reply:
fair, forgot about that - your original code is better.